### PR TITLE
[#IP-362] Added a codec to support conversion from string to boolean

### PR DESCRIPTION
**Pull request by michaeldisaro**
We need to convert a string to boolean when dealing with request's parameters.

Added tests for codec.

### CODE REVIEW

File: src/booleans.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+      E.chain(s =>
+        s === "true"
+          ? t.success(true)
+          : s === "false"
+          ? t.success(false)
+          : t.failure(u, c)
+      )
+    ),
+  String
+);
```
Comment (balanza) [inline src/booleans.ts:26]: We can avoid the string validation per se as `s === 'true'` implies `s` is a string. Also, by being explicit with the `I` (input) type, we can avoid non-string to even be passed to the decoder

```suggestion
export const BooleanFromString: BooleanFromString = new t.Type<
  boolean,
  string,
  string
>(
  "BooleanFromString",
  t.boolean.is,
  (s, c) =>
    E.chain(s =>
      s === "true"
        ? t.success(true)
        : s === "false"
        ? t.success(false)
        : t.failure(u, c)
    ),
  String
);
```

Comment (michaeldisaro) [reply src/booleans.ts:26]: Yeah, we just got gcanti code. We tried to import io-ts-types but that library was breaking build, so we just copied what we needed.  Anyway, your comment seems legit.

Comment (gquadrati) [reply src/booleans.ts]: In this case we nee to remove `E.chain` too.  Also, it's not a `BooleanFromString` anymore

Comment (michaeldisaro) [reply src/booleans.ts]: Done in https://github.com/pagopa/io-ts-commons/pull/367/commits/2c9ed4840e10021078ef139f8410f67c38e4e2f6

**Review by balanza — APPROVED** [review]

**Review by gquadrati — APPROVED** [review]

**Pull request merged**